components/core/src/LibarchiveReader.cpp

```diff
@@ -186,53 +180,31 @@ ErrorCode LibarchiveReader::libarchive_read_callback (const void** buffer, size_
     if (false == m_is_opened_by_libarchive) {
         return ErrorCode_NotInit;
     }
-
-    if (false == m_initial_buffer_content_exhausted) {
-        *buffer = m_buffer.data();
-        num_bytes_read = m_buffer.size();
-        m_initial_buffer_content_exhausted = true;
-    } else {
-        constexpr size_t cTargetBufferLength = 4096;
-        m_buffer.resize(cTargetBufferLength);
-        auto error_code = m_file_reader->try_read(m_buffer.data(), cTargetBufferLength, num_bytes_read);
-        if (ErrorCode_Success != error_code) {
-            return error_code;
-        }
-        if (num_bytes_read < cTargetBufferLength) {
-            m_buffer.resize(num_bytes_read);
-        }
-    }
-
-    return ErrorCode_Success;
-}
-
-ErrorCode LibarchiveReader::libarchive_skip_callback (off_t num_bytes_to_skip, size_t& num_bytes_skipped) {
-    // Get current position
-    size_t pos;
-    auto error_code = m_file_reader->try_get_pos(pos);
-    if (ErrorCode_Success != error_code) {
-        return error_code;
-    }
-
-    // Calculate desired position, ensuring its within the file
-    size_t desired_pos = pos + num_bytes_to_skip;
-    struct stat stat_buffer = {};
-    error_code = m_file_reader->try_fstat(stat_buffer);
+    constexpr size_t cTargetBufferLength = 4096;
+    m_buffer.resize(cTargetBufferLength);
+    auto error_code = m_file_reader->try_read(m_buffer.data(), cTargetBufferLength, num_bytes_read);
     if (ErrorCode_Success != error_code) {
         return error_code;
     }
-    if (desired_pos > stat_buffer.st_size) {
-        desired_pos = stat_buffer.st_size;
+    if (num_bytes_read < cTargetBufferLength) {
+        m_buffer.resize(num_bytes_read);
     }
+    *buffer = m_buffer.data();
+    return ErrorCode_Success;
+}
 
-    // Seek to desired position
-    error_code = m_file_reader->try_seek_from_begin(desired_pos);
-    if (ErrorCode_Success != error_code) {
+ErrorCode LibarchiveReader::libarchive_skip_callback (off_t num_bytes_to_skip,
```

Comment:
We don't need this callback anymore since we're not doing anything more efficient than reading (which is what libarchive will do when there's no skip callback).